### PR TITLE
#470 inline initialization of fields is forbidden

### PR DESCRIPTION
**Pull request by krzyk**
#470 
* created two test cases (with no constrcutor, and with one)
* implemented check using PMD xpath

### CODE REVIEW

**Comment by davvd** [comment]
@krzyk Thanks, I will ask someone to do a code review here, thanks for your pull request

**Comment by davvd** [comment]
@carlosmiranda it's yours, please review

File: qulice-pmd/src/main/resources/com/qulice/pmd/ruleset.xml [hunk 1]
```diff
@@ -96,4 +96,29 @@
...
+          language="java"
+          class="net.sourceforge.pmd.lang.rule.XPathRule">
+        <description>
+            Avoid doing field initialization outside class constructor.
+            Such initialization is allowed only when there is no constructor.
+        </description>
+        <properties>
+            <property name="xpath">
+                <value>
+                    <!--
```
Comment (carlosmiranda) [inline qulice-pmd/src/main/resources/com/qulice/pmd/ruleset.xml]: @krzyk Let's remove this commented-out section.

**Comment by carlosmiranda** [comment]
@krzyk looks good, just one comment from me.

**Comment by krzyk** [comment]
@carlosmiranda fixed

**Comment by carlosmiranda** [comment]
@krzyk thanks!

**Comment by carlosmiranda** [comment]
@rultor merge please

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@carlosmiranda paid, thanks, added **18 mins** to your account, payment `67436939`, 48 hours and 9 mins was spent

review comments (c=3) added as a bonus

+18 added to your rating, current score is: [+3830](http://www.netbout.com/b/35023?open=rating)

**Comment by davvd** [comment]
@rultor deploy

**Comment by pinaf** [comment]
@krzyk this does not differentiate between static and non-static fields, correct?

**Comment by krzyk** [comment]
@pinaf I don't know yet, but I assume you asked because you have seen the problem with statics?

**Comment by pinaf** [comment]
@krzyk yup, I get a `PMD.ConstructorShouldDoInitialization` error when a static field is the only one being initialized on declaration.

**Pull request merged**